src/proposals/MultisigProposal.sol

```diff
@@ -49,16 +77,34 @@ abstract contract MultisigProposal is Proposal {
     function _simulateActions(address multisig) internal {
         vm.startPrank(multisig);
 
-        /// this is a hack because multisig execTransaction requires owners signatures
-        /// so we cannot simulate it exactly as it will be executed on mainnet
-        vm.etch(multisig, Constants.MULTISEND_BYTECODE);
-
         bytes memory data = getCalldata();
 
-        multisig.functionCall(data);
+        // this is a hack because multisig execTransaction requires owners signatures
+        // so the SAFE_CREATION_BYTECODE override the checkNSignatures function
+        bytes memory bytecode = Constants.SAFE_CREATION_BYTECODE;
+        address addr;
+        /// @solidity memory-safe-assembly
+        assembly {
+            addr := create(0, add(bytecode, 0x20), mload(bytecode))
+        }
```

Comment:
I'd prefer etching runtime bytecode of this safe with the checkNSignatures function stubbed out instead of needing to create it every time

Reply:
then no need to deploy

Reply:
How that would work without deploying. Etching only works with deployed bytecode, right?

Reply:
grab the runtime bytecode from an instance you deploy on your local machine in a foundry test, log it out to the terminal, copy and paste that here, rename from SAFE_CREATION_BYTECODE to RUNTIME_BYTECODE